### PR TITLE
Fix files_sharing app LoadSidebar event

### PR DESCRIPTION
**Pull request by skjnldsv**
Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

### CODE REVIEW

File: apps/files_sharing/lib/AppInfo/Application.php [hunk 1]
```diff
@@ -161,21 +177,98 @@ function() use ($c) {
...
+			\OCP\Util::addScript('files_sharing', 'dist/collaboration');
+		});
+	}
+
+	protected function setupSharingMenus() {
+		$config = \OC::$server->getConfig();
+		$userSession = \OC::$server->getUserSession();
+		$l = \OC::$server->getL10N('files_sharing');
+
+		if ($config->getAppValue('core', 'shareapi_enabled', 'yes') === 'yes') {
```
Comment (kesselb) [inline apps/files_sharing/lib/AppInfo/Application.php]: I think it's much easier to read if you check for `'no'` here and just `return;` in case.

Comment (kesselb) [reply apps/files_sharing/lib/AppInfo/Application.php]:
```
Index: apps/files_sharing/lib/AppInfo/Application.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/files_sharing/lib/AppInfo/Application.php	(revision a1dbd9548d461c2e6647872c71dfa4bdc9d2e0f8)
+++ apps/files_sharing/lib/AppInfo/Application.php	(date 1572975007209)
@@ -208,67 +208,67 @@
 		$userSession = \OC::$server->getUserSession();
 		$l = \OC::$server->getL10N('files_sharing');
 
-		if ($config->getAppValue('core', 'shareapi_enabled', 'yes') === 'yes') {
+		if ($config->getAppValue('core', 'shareapi_enabled', 'yes') === 'no') {
+			return;
+		}
 
-			$sharingSublistArray = [];
-		
-			if (\OCP\Util::isSharingDisabledForUser() === false) {
-				array_push($sharingSublistArray, [
-					'id' => 'sharingout',
-					'appname' => 'files_sharing',
-					'script' => 'list.php',
-					'order' => 16,
-					'name' => $l->t('Shared with others'),
-				]);
-			}
-		
-			array_push($sharingSublistArray, [
-				'id' => 'sharingin',
-				'appname' => 'files_sharing',
-				'script' => 'list.php',
-				'order' => 15,
-				'name' => $l->t('Shared with you'),
-			]);
-		
-			if (\OCP\Util::isSharingDisabledForUser() === false) {
-				// Check if sharing by link is enabled
-				if ($config->getAppValue('core', 'shareapi_allow_links', 'yes') === 'yes') {
-					array_push($sharingSublistArray, [
-						'id' => 'sharinglinks',
-						'appname' => 'files_sharing',
-						'script' => 'list.php',
-						'order' => 17,
-						'name' => $l->t('Shared by link'),
-					]);
-				}
-			}
-		
-			array_push($sharingSublistArray, [
-				'id' => 'deletedshares',
-				'appname' => 'files_sharing',
-				'script' => 'list.php',
-				'order' => 19,
-				'name' => $l->t('Deleted shares'),
-			]);
-		
-			// show_Quick_Access stored as string
-			$user = $userSession->getUser();
-			$defaultExpandedState = true;
-			if ($user instanceof \OCP\IUser) {
-				$defaultExpandedState = $config->getUserValue($userSession->getUser()->getUID(), 'files', 'show_sharing_menu', '0') === '1';
-			}
-		
-			\OCA\Files\App::getNavigationManager()->add([
-				'id' => 'shareoverview',
-				'appname' => 'files_sharing',
-				'script' => 'list.php',
-				'order' => 18,
-				'name' => $l->t('Shares'),
-				'classes' => 'collapsible',
-				'sublist' => $sharingSublistArray,
-				'expandedState' => 'show_sharing_menu'
-			]);
-		}
-		
+		$sharingSublistArray = [];
+
+		if (\OCP\Util::isSharingDisabledForUser() === false) {
+			$sharingSublistArray[] = [
+				'id' => 'sharingout',
+				'appname' => 'files_sharing',
+				'script' => 'list.php',
+				'order' => 16,
+				'name' => $l->t('Shared with others'),
+			];
+		}
+
+		$sharingSublistArray[] = [
+			'id' => 'sharingin',
+			'appname' => 'files_sharing',
+			'script' => 'list.php',
+			'order' => 15,
+			'name' => $l->t('Shared with you'),
+		];
+
+		if (\OCP\Util::isSharingDisabledForUser() === false) {
+			// Check if sharing by link is enabled
+			if ($config->getAppValue('core', 'shareapi_allow_links', 'yes') === 'yes') {
+				$sharingSublistArray[] = [
+					'id' => 'sharinglinks',
+					'appname' => 'files_sharing',
+					'script' => 'list.php',
+					'order' => 17,
+					'name' => $l->t('Shared by link'),
+				];
+			}
+		}
+
+		$sharingSublistArray[] = [
+			'id' => 'deletedshares',
+			'appname' => 'files_sharing',
+			'script' => 'list.php',
+			'order' => 19,
+			'name' => $l->t('Deleted shares'),
+		];
+
+		// show_Quick_Access stored as string
+		$user = $userSession->getUser();
+		$defaultExpandedState = true;
+		if ($user instanceof \OCP\IUser) {
+			$defaultExpandedState = $config->getUserValue($userSession->getUser()->getUID(), 'files', 'show_sharing_menu', '0') === '1';
+		}
+
+		\OCA\Files\App::getNavigationManager()->add([
+			'id' => 'shareoverview',
+			'appname' => 'files_sharing',
+			'script' => 'list.php',
+			'order' => 18,
+			'name' => $l->t('Shares'),
+			'classes' => 'collapsible',
+			'sublist' => $sharingSublistArray,
+			'expandedState' => 'show_sharing_menu'
+		]);
 	}
 }
```

I haven't tested it :see_no_evil:

Comment (skjnldsv) [reply apps/files_sharing/lib/AppInfo/Application.php]: Done :)

**Comment by skjnldsv** [comment]
All fixed! Please review! :)

**Review by ChristophWurst — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Comment by juliushaertl** [comment]
Tests are not happy:
```
Error: Call to protected method OCA\Files_Sharing\AppInfo\Application::registerMountProviders() from context 'TrashbinTest' in /drone/src/apps/files_trashbin/tests/TrashbinTest.php:71
```

jsunit/phan tests are marked as canceled, maybe just a temporary issue.

**Comment by skjnldsv** [comment]
Hopefully lasts fixes for tests to pass! :sob:

**Comment by skjnldsv** [comment]
Tests passed :rocket:

**Review by rullzer — APPROVED** [review]

File: apps/files_sharing/lib/AppInfo/Application.php [hunk 2]
```diff
@@ -164,26 +180,37 @@ function() use ($c) {
...
-		$server = $this->getContainer()->query('ServerContainer');
-		$mountProviderCollection = $server->getMountProviderCollection();
+	protected function registerMountProviders(IMountProviderCollection $mountProviderCollection) {
 		$mountProviderCollection->registerProvider($this->getContainer()->query('MountProvider'));
 		$mountProviderCollection->registerProvider($this->getContainer()->query('ExternalMountProvider'));
 	}
 
-	public function register(): void {
-		$dispatcher = $this->getContainer()->getServer()->getEventDispatcher();
+	protected function registerEventsScripts(IEventDispatcher $dispatcher) {
```
Comment (MorrisJobke) [inline apps/files_sharing/lib/AppInfo/Application.php:205]: This is where we broke the dispatch of addListener in line 214, where GenericEvent is type hinted in the callback, right @rullzer? That's what Psalm found in https://github.com/nextcloud/server/pull/23742/files#diff-f24715432b9a8ad98f1ff9c441096d8afd1ed1bcfb43f04a04ad2894e050e2a8R156 


Nice catch by psalm I would say - cc @ChristophWurst @kesselb

**Pull request merged**